sdk/openai/Azure.AI.OpenAI/src/Custom/Choice.Serialization.cs

```diff
@@ -16,10 +16,10 @@ public partial class Choice
     {
         internal static Choice DeserializeChoice(JsonElement element)
```

Comment:
Can all this custom implementation be deleted and let it be auto-generated?

Reply:
Unfortunately, no--though I should (and will now) add a comment into the custom code to clarify why.

Right now, the generated version yields this for some requests:
```
  Message: 
System.Text.Json.JsonException : A property 'logprobs' defined as non-nullable but received as null from the service. This exception only happens in DEBUG builds of the library and would be ignored in the release build

  Stack Trace: 
JsonElementExtensions.ThrowNonNullablePropertyIsNull(JsonProperty property) line 105
Choice.DeserializeChoice(JsonElement element) line 42
```

This may be a generalized version of the "nullable float" problem that was addressed, though in this case making `logprobs?` in `Choice` `CompletionsLogProbabilityModel | null` doesn't seem to address it.

Reply:
Looking at the cadl.json that is produced during code generation, I see the `LogProbabilityModel` property of the `Choice` model and that its `IsNullable` value is `true`. Therefore, I think this is probably a bug in the emitter that is causing it to not respect this value.

@m-nash: What do you think? I'm thinking we keep the custom serialization for now, and I'll file an issue for this in the [autorest.csharp repo](https://github.com/Azure/autorest.csharp). Do you know if it needs any special tags or anything in particular?

Reply:
Thanks, @m-nash. Updated to latest and confirmed that it works!